lib/Actions.js

```diff
@@ -10,10 +10,17 @@ function Actions (options) {
 Actions.prototype = {
     TYPES: {
         CLICK: 'click',
+        MOUSE_DOWN: 'mousedown',
+        MOUSE_UP: 'mouseup',
         WAIT: 'wait',
+        WAIT_FOR_VISIBLE: 'waitForVisible',
+        WAIT_FOR_PATTERN: 'waitForPattern',
+        WAIT_FOR_PAGE: 'waitForPage',
         TYPE: 'type',
         CONDITION: 'conditionalActions',
-        EXIST: 'exist'
+        EXIST: 'exist',
+        BACK: 'back',
+        PROVIDE_COLLECTION: 'provideCollection'
```

Comment:
Sometimes we do not know what are rules for parsing before actions are performed. Also we have conditional actions and last action can return collection to parse, this collection will be attached to the rule which actions are performed for.